### PR TITLE
Update CHANGELOG.md and README.md for 0.4.0

### PR DESCRIPTION
**Pull request by kevinoid**
I really like the changes you made in version 0.4.0, but found it difficult to use since the changes are undocumented.  This PR updates `CHANGELOG.md` and `README.md` with current information, as best I can deduce it.

Thanks for considering,
Kevin

P.S.  You also may want to tag the 0.4.0 commit in your repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +61,7 @@ suppose('npm', ['init'])
 })
 ```
 
-Always follow an `.on()` with a `.respond()` and then finish with a `.end()`.
+Always follow an `.when()` with a `.respond()` and then finish with a `.end()`.
```
Comment (piranna) [inline README.md]: That's not true anymore, you can interleave how much `.when()` and `.respond()` as you want.

**Comment by kevinoid** [comment]
Ok.  I've updated the PR to fix the issue that you mentioned about any number of `.respond()` calls after `.when()`.  See what you think.

**Comment by piranna** [comment]
Ok, now it's good :-)

**Comment by kevinoid** [comment]
Great!  Thanks again!

**Pull request merged**